tracer/test/Datadog.Trace.Tests/Configuration/DynamicConfigurationTests.cs

```diff
@@ -42,27 +42,69 @@ public void ApplyConfigurationTwice()
         {
             var tracer = TracerManager.Instance;
 
-            DynamicConfigurationManager.OnlyForTests_ApplyConfiguration(CreateConfig((ConfigurationKeys.GlobalSamplingRate, "0.4")));
+            DynamicConfigurationManager.OnlyForTests_ApplyConfiguration(CreateConfig(("tracing_sampling_rate", "0.4")));
 
             var newTracer = TracerManager.Instance;
 
             newTracer.Should().NotBeSameAs(tracer);
 
-            DynamicConfigurationManager.OnlyForTests_ApplyConfiguration(CreateConfig((ConfigurationKeys.GlobalSamplingRate, "0.4")));
+            DynamicConfigurationManager.OnlyForTests_ApplyConfiguration(CreateConfig(("tracing_sampling_rate", "0.4")));
 
             TracerManager.Instance.Should().BeSameAs(newTracer);
         }
 
+        [Fact]
+        public void ApplyTagsToDirectLogs()
+        {
+            var tracerSettings = new TracerSettings();
+
+            tracerSettings.GlobalTagsInternal.Add("key1", "value1");
+
+            TracerManager.ReplaceGlobalManager(new ImmutableTracerSettings(tracerSettings), TracerManagerFactory.Instance);
+
+            TracerManager.Instance.DirectLogSubmission.Formatter.Tags.Should().Be("key1:value1");
+
+            DynamicConfigurationManager.OnlyForTests_ApplyConfiguration(CreateConfig(("tracing_tags", "['key2:value2']")));
+
+            TracerManager.Instance.DirectLogSubmission.Formatter.Tags.Should().Be("key2:value2");
+        }
+
+        [Fact]
+        public void DoesNotOverrideDirectLogsTags()
+        {
+            var tracerSettings = new TracerSettings();
+            tracerSettings.LogSubmissionSettings.DirectLogSubmissionGlobalTags.Add("key1", "value1");
+            tracerSettings.LogSubmissionSettings.DirectLogSubmissionEnabledIntegrations.Add("test");
+
+            tracerSettings.GlobalTagsInternal.Add("key2", "value2");
+
+            TracerManager.ReplaceGlobalManager(new ImmutableTracerSettings(tracerSettings), TracerManagerFactory.Instance);
+
+            TracerManager.Instance.DirectLogSubmission.Formatter.Tags.Should().Be("key1:value1");
+
+            DynamicConfigurationManager.OnlyForTests_ApplyConfiguration(CreateConfig(("tracing_tags", "['key3:value3']")));
+
+            TracerManager.Instance.DirectLogSubmission.Formatter.Tags.Should().Be("key1:value1");
```

Comment:
I mean, this does raise the question of whether remote config _should_ overwrite these, but I think this behaviour is fine

Reply:
+1 this behavior seems fine to me